python/lsst/meas/algorithms/objectSizeStarSelector.py

```diff
@@ -430,21 +428,35 @@ def selectStars(self, exposure, sourceCat, matches=None):
         centers, clusterId = _kcenters(width, nCluster=4, useMedian=True,
                                        widthStdAllowed=self.config.widthStdAllowed)
 
-        if display and plotMagSize and pyplot:
-            fig = plot(mag, width, centers, clusterId,
-                       magType=self.config.sourceFluxField.split(".")[-1].title(),
-                       marker="+", markersize=3, markeredgewidth=None, ltype=':', clear=True)
-        else:
-            fig = None
-
         clusterId = _improveCluster(width, centers, clusterId,
                                     nsigma=self.config.nSigmaClip,
                                     widthStdAllowed=self.config.widthStdAllowed)
 
-        if display and plotMagSize and pyplot:
-            plot(mag, width, centers, clusterId, marker="x", markersize=3, markeredgewidth=None, clear=False)
+        if display and plotMagSize:
+            self._plot(display, displayExposure, exposure, plotMagSize, mag, width,
+                       centers, clusterId, sourceCat, good)
 
         stellar = (clusterId == 0)
+        starCat = SourceCatalog(sourceCat.table)
+        goodSources = [s for g, s in zip(good, sourceCat) if g]
+        for isStellar, source in zip(stellar, goodSources):
+            if isStellar:
+                starCat.append(source)
+
+        return Struct(
+            starCat=starCat,
+        )
+
+    def _plot(self, display, displayExposure, exposure, plotMagSize, mag, width,
```

Comment:
Hmm... in the interest of not mucking with this mess any more than I have to, I'm going to get rid of the private _plot() method I made (which may have changed how the plots were created), and just keep the moved matplotlib import.